### PR TITLE
Extend TabWidget's style sheet options (#6730)

### PR DESCRIPTION
**Pull request by michaelgregorius**
Extend the `TabWidget` class so that the text color of the selected tab can be set in the style sheet. Adjust the paint method to make use of the new property.

Adjust the default style sheet as follows:
* Background color of the selected tab is the green of the knobs
* Text color of the selected tab is full on white

Adjust the classic style sheet in such a way that nothing changes, i.e. the text colors of the selected tab and the other ones are the same.

Fixes https://github.com/LMMS/lmms/issues/6730

### CODE REVIEW

**Comment by zonkmachine** [comment]
Before
tabcolorbefore

After
tabcolor

**Review by zonkmachine — APPROVED** [review]
Tested, looks good!

This improves visibility. I think the tab color of the default theme could be a bit brighter still but this is acceptable to me.

**Comment by anytizer** [comment]
Looking distinct now. But still, if the text were as green as the LED color, would be even better, instead of white.
#6730 has greener suggestions (black or transparent background with text on green).

**Comment by michaelgregorius** [comment]
I have experimented a bit and set the highlighted text color of the tab to a light green text as proposed by @anytizer. Here's how that looks with the changed dark green background and the unchanged background that the current master uses:

6730-TabWidgetChanges-GreenTextHighlight 6730-TabWidgetChanges-GreenTextOldBackground

The grey background is a bit more neutral and does not bite with any other potential colors in the content area. On the other hand it is not as obvious because it is rather subtle.

File: include/TabWidget.h [hunk 1]
```diff
@@ -59,6 +59,7 @@ class TabWidget : public QWidget
 	Q_PROPERTY( QColor tabText READ tabText WRITE setTabText)
 	Q_PROPERTY( QColor tabTitleText READ tabTitleText WRITE setTabTitleText)
 	Q_PROPERTY( QColor tabSelected READ tabSelected WRITE setTabSelected)
+	Q_PROPERTY( QColor tabTextSelected READ tabTextSelected WRITE setTabTextSelected)
```
Comment (sakertooth) [inline include/TabWidget.h:62]:
```suggestion
	Q_PROPERTY(QColor tabTextSelected READ tabTextSelected WRITE setTabTextSelected)
```

File: include/TabWidget.h [hunk 2]
```diff
@@ -68,6 +69,8 @@ class TabWidget : public QWidget
 	void setTabTitleText( const QColor & c );
 	QColor tabSelected() const;
 	void setTabSelected( const QColor & c );
+	QColor tabTextSelected() const;
+	void setTabTextSelected( const QColor & c );
```
Comment (sakertooth) [inline include/TabWidget.h:73]:
```suggestion
	void setTabTextSelected(const QColor & c);
```

File: src/gui/widgets/TabWidget.cpp [hunk 3]
```diff
@@ -48,6 +48,7 @@ TabWidget::TabWidget(const QString & caption, QWidget * parent, bool usePixmap,
 	m_tabText( 0, 0, 0 ),
 	m_tabTitleText( 0, 0, 0 ),
 	m_tabSelected( 0, 0, 0 ),
+	m_tabTextSelected( 0, 0, 0 ),
```
Comment (sakertooth) [inline src/gui/widgets/TabWidget.cpp:51]:
```suggestion
	m_tabTextSelected(0, 0, 0),
```

File: src/gui/widgets/TabWidget.cpp [hunk 4]
```diff
@@ -274,10 +275,15 @@ void TabWidget::paintEvent( QPaintEvent * pe )
 			if( it.key() == m_activeTab )
 			{
 				p.fillRect( tab_x_offset, 2, ( *it ).nwidth - 6, m_tabbarHeight - 4, tabSelected() );
+				p.setPen(tabTextSelected());
+				p.drawText( tab_x_offset + 3, m_tabheight + 1, ( *it ).name );
+			}
+			else
+			{
+				// Draw text
+				p.setPen( tabText() );
```
Comment (sakertooth) [inline src/gui/widgets/TabWidget.cpp:284]:
```suggestion
				p.setPen(tabText());
```

File: src/gui/widgets/TabWidget.cpp [hunk 5]
```diff
@@ -274,10 +275,15 @@ void TabWidget::paintEvent( QPaintEvent * pe )
 			if( it.key() == m_activeTab )
 			{
 				p.fillRect( tab_x_offset, 2, ( *it ).nwidth - 6, m_tabbarHeight - 4, tabSelected() );
+				p.setPen(tabTextSelected());
+				p.drawText( tab_x_offset + 3, m_tabheight + 1, ( *it ).name );
```
Comment (sakertooth) [inline src/gui/widgets/TabWidget.cpp:279]:
```suggestion
				p.drawText(tab_x_offset + 3, m_tabheight + 1, it->name);
```

File: src/gui/widgets/TabWidget.cpp [hunk 6]
```diff
@@ -274,10 +275,15 @@ void TabWidget::paintEvent( QPaintEvent * pe )
...
+			}
+			else
+			{
+				// Draw text
+				p.setPen( tabText() );
+				p.drawText( tab_x_offset + 3, m_tabheight + 1, ( *it ).name );
```
Comment (sakertooth) [inline src/gui/widgets/TabWidget.cpp:285]:
```suggestion
				p.drawText(tab_x_offset + 3, m_tabheight + 1, it->name);
```

File: src/gui/widgets/TabWidget.cpp [hunk 7]
```diff
@@ -392,6 +398,18 @@ void TabWidget::setTabSelected( const QColor & c )
...
 }
 
+// Return the text color of the selected tab
+QColor TabWidget::tabTextSelected() const
+{
+	return m_tabTextSelected;
+}
+
+// Set the text color of the selected tab
+void TabWidget::setTabTextSelected( const QColor & c )
```
Comment (sakertooth) [inline src/gui/widgets/TabWidget.cpp:408]:
```suggestion
void TabWidget::setTabTextSelected(const QColor & c)
```

**Review by sakertooth — APPROVED** [review]

File: include/TabWidget.h (hunk 1 above)
Comment (PhysSong) [reply include/TabWidget.h:62]: Although the rest use `READ` and `WRITE`, it's also possible to use `MEMBER`:
```suggestion
	Q_PROPERTY(QColor tabTextSelected MEMBER m_tabTextSelected)
```

Comment (michaelgregorius) [reply include/TabWidget.h]: Spaces fixed with commit 0fa18d405e7 where I have fixed style issues for all of `TabWidget`. I have kept the `READ` and `WRITE` versions due to consistency with the other properties.

File: src/gui/widgets/TabWidget.cpp (hunk 3 above)
Comment (michaelgregorius) [reply src/gui/widgets/TabWidget.cpp]: Spaces fixed with commit 0fa18d405e7 where I have fixed style issues for all of `TabWidget`.

File: include/TabWidget.h (hunk 2 above)
Comment (michaelgregorius) [reply include/TabWidget.h]: Spaces fixed with commit 0fa18d405e7 where I have fixed style issues for all of `TabWidget`.

File: src/gui/widgets/TabWidget.cpp (hunk 4 above)
Comment (michaelgregorius) [reply src/gui/widgets/TabWidget.cpp]: Spaces fixed with commit 0fa18d405e7 where I have fixed style issues for all of `TabWidget`.

File: src/gui/widgets/TabWidget.cpp (hunk 7 above)
Comment (michaelgregorius) [reply src/gui/widgets/TabWidget.cpp]: Spaces fixed with commit 0fa18d405e7 where I have fixed style issues for all of `TabWidget`.

File: src/gui/widgets/TabWidget.cpp (hunk 5 above)
Comment (michaelgregorius) [reply src/gui/widgets/TabWidget.cpp]: Spaces fixed with commit 0fa18d405e7 where I have fixed style issues for all of `TabWidget`. Repeated dereferencing of iterators has been fixed with commit 90affa202b4 by introducing variables with speaking names.

File: src/gui/widgets/TabWidget.cpp (hunk 6 above)
Comment (michaelgregorius) [reply src/gui/widgets/TabWidget.cpp]: Spaces fixed with commit 0fa18d405e7 where I have fixed style issues for all of `TabWidget`. Repeated dereferencing of iterators has been fixed with commit 90affa202b4 by introducing variables with speaking names.

**Pull request merged**